src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/courant_dev/parity_test.ipynb

```diff
@@ -0,0 +1,428 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Testing Courant capability in Muskingum-Cunge FORTRAN module\n",
+    "The Courant number is an important diagnostic metric for river routing models, defined as the ratio of the kinematic wave celerity to numerical celerity (the grid ratio Δx/Δt). According to Ponce, the Muskingum-Cunge routing method is a reasonable representation of the physical prototype if the Courant number is close to 1 (among other parametric limitations). Therefore, to assess the numerical accuracy of Musking-Cunge routing simulations across the NHDPlus, it is adventageous to  output the Courant condition for each segment and timestep of a routing simulation. \n",
+    "\n",
+    "Here, we test an additional subroutine added to the Muskingum-Cunge FORTRAN module to calculate Courant condition and kinematic celerity upon every call of the model. Additionally the MC module is updated to return the X parameter, another valuable diagnostic. This parity tests for any unintended changes in simulated flow, velocity, and depth brought about by the newly implemented subroutine. Additionally, the Courant number and kinematic celerity values calculated by the new subroutine are compared against Courant number and kinematic celerity results calculated with a simple Python function, using outputs from the original MC module. \n",
+    "\n",
+    "A few important notes before we dive in. \n",
+    "1. The modifications we are testing on top of `t-route/src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/MCsingleSegStime_f2py_NOLOOP.f90`.  This file contains a new subroutine to calculate and output the Courant number and kinematic celerity.\n",
+    "2. The original (prior to any changes) version of this file is copied here: `t-route/src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/courant_dev/MCsingleSegStime_f2py_NOLOOP-original.f90`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "import os\n",
+    "import subprocess\n",
+    "import csv\n",
+    "import numpy as np\n",
+    "import time\n",
+    "\n",
+    "try:\n",
+    "    import google.colab\n",
+    "\n",
+    "    ENV_IS_CL = True\n",
+    "    root = r\"/content/t-route\"\n",
+    "    subprocess.run([\"git\", \n",
+    "                    \"clone\", \n",
+    "                    \"-b\",\n",
+    "                    \"courant\",\n",
+    "                    \"https://github.com/awlostowski-noaa/t-route.git\"])\n",
```

Comment:
We'll want this to point to the NOAA-OWP master when we make the final commit/merge.

Reply:
noted as TODO in the notebook. Will circle back once we are ready to merge.